vendor/github.com/opencontainers/image-spec/specs-go/v1/config.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+// ImageConfig defines the execution parameters which should be used as a base when running a container using an image.
+type ImageConfig struct {
+	// User defines the username or UID which the process in the container should run as.
+	User string `json:"User,omitempty"`
+
+	// Memory defines the memory limit.
+	Memory int64 `json:"Memory,omitempty"`
+
+	// MemorySwap defines the total memory usage limit (memory + swap).
+	MemorySwap int64 `json:"MemorySwap,omitempty"`
+
+	// CPUShares is the CPU shares (relative weight vs. other containers).
+	CPUShares int64 `json:"CpuShares,omitempty"`
```

Comment:
@runcom We really can't tag a v1 because it is simply not ready. Basic stuff like this is not even figured out and I keep finding junk that is just shoved in. /rant